// next@15.2.0-canary.27/dist/client/components/react-dev-overlay/_experimental/internal/components/errors/dialog/dialog.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
0 && (module.exports = {
    DIALOG_STYLES: null,
    ErrorOverlayDialog: null
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    DIALOG_STYLES: function() {
        return DIALOG_STYLES;
    },
    ErrorOverlayDialog: function() {
        return ErrorOverlayDialog;
    }
});
const _tagged_template_literal_loose = require("@swc/helpers/_/_tagged_template_literal_loose");
const _jsxruntime = require("react/jsx-runtime");
const _dialog = require("../../dialog/dialog");
const _nooptemplate = require("../../../helpers/noop-template");
function _templateObject() {
    const data = _tagged_template_literal_loose._([
        "\n  .error-overlay-dialog {\n    background: var(--color-background-100);\n    border: 1px solid var(--color-gray-400);\n    border-radius: var(--rounded-xl);\n    box-shadow: var(--shadow-md);\n  }\n\n  .nextjs-error-overlay-dialog-turbopack-background {\n    border: 1px solid transparent;\n    background:\n      linear-gradient(var(--color-background-100), var(--color-background-100))\n        padding-box,\n      linear-gradient(\n          to right top,\n          var(--color-gray-400) 75%,\n          var(--color-turbopack-border-blue) 87.5%,\n          var(--color-turbopack-border-red) 100%\n        )\n        border-box;\n  }\n"
    ]);
    _templateObject = function() {
        return data;
    };
    return data;
}
function ErrorOverlayDialog(param) {
    let { isTurbopack, children, onClose } = param;
    return /*#__PURE__*/ (0, _jsxruntime.jsx)(_dialog.Dialog, {
        type: "error",
        "aria-labelledby": "nextjs__container_errors_label",
        "aria-describedby": "nextjs__container_errors_desc",
        onClose: onClose,
        className: "error-overlay-dialog " + (isTurbopack ? 'nextjs-error-overlay-dialog-turbopack-background' : ''),
        children: children
    });
}
const DIALOG_STYLES = (0, _nooptemplate.noop)(_templateObject());

if ((typeof exports.default === 'function' || (typeof exports.default === 'object' && exports.default !== null)) && typeof exports.default.__esModule === 'undefined') {
  Object.defineProperty(exports.default, '__esModule', { value: true });
  Object.assign(exports.default, exports);
  module.exports = exports.default;
}
