// next@15.2.0-canary.27/dist/client/components/react-dev-overlay/_experimental/internal/components/errors/error-overlay-floating-header/error-overlay-floating-header.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
0 && (module.exports = {
    ErrorOverlayFloatingHeader: null,
    styles: null
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    ErrorOverlayFloatingHeader: function() {
        return ErrorOverlayFloatingHeader;
    },
    styles: function() {
        return styles;
    }
});
const _tagged_template_literal_loose = require("@swc/helpers/_/_tagged_template_literal_loose");
const _jsxruntime = require("react/jsx-runtime");
const _erroroverlaypagination = require("../error-overlay-pagination/error-overlay-pagination");
const _versionstalenessinfo = require("../../version-staleness-info/version-staleness-info");
const _nooptemplate = require("../../../helpers/noop-template");
function _templateObject() {
    const data = _tagged_template_literal_loose._([
        "\n  .error-overlay-floating-header {\n    display: flex;\n    justify-content: space-between;\n    align-items: center;\n\n    width: 100%;\n    position: fixed;\n    transform: translateY(calc(-1 * var(--size-10_5)));\n\n    outline: none;\n\n    @media (min-width: 576px) {\n      max-width: 540px;\n    }\n\n    @media (min-width: 768px) {\n      max-width: 720px;\n    }\n\n    @media (min-width: 992px) {\n      max-width: 960px;\n    }\n  }\n"
    ]);
    _templateObject = function() {
        return data;
    };
    return data;
}
function ErrorOverlayFloatingHeader(param) {
    let { readyErrors, activeIdx, setActiveIndex, versionInfo, isTurbopack } = param;
    return /*#__PURE__*/ (0, _jsxruntime.jsxs)("div", {
        className: "error-overlay-floating-header",
        children: [
            /*#__PURE__*/ (0, _jsxruntime.jsx)(_erroroverlaypagination.ErrorOverlayPagination, {
                readyErrors: readyErrors != null ? readyErrors : [],
                activeIdx: activeIdx != null ? activeIdx : 0,
                onActiveIndexChange: setActiveIndex != null ? setActiveIndex : ()=>{}
            }),
            /*#__PURE__*/ (0, _jsxruntime.jsx)(_versionstalenessinfo.VersionStalenessInfo, {
                versionInfo: versionInfo,
                isTurbopack: isTurbopack
            })
        ]
    });
}
const styles = (0, _nooptemplate.noop)(_templateObject());

if ((typeof exports.default === 'function' || (typeof exports.default === 'object' && exports.default !== null)) && typeof exports.default.__esModule === 'undefined') {
  Object.defineProperty(exports.default, '__esModule', { value: true });
  Object.assign(exports.default, exports);
  module.exports = exports.default;
}
