// next@15.2.0-canary.27/dist/client/components/react-dev-overlay/_experimental/app/react-dev-overlay.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
Object.defineProperty(exports, "default", {
    enumerable: true,
    get: function() {
        return ReactDevOverlay;
    }
});
const _jsxruntime = require("react/jsx-runtime");
const _react = require("react");
const _errorboundary = require("./error-boundary");
const _shadowportal = require("../internal/components/shadow-portal");
const _base = require("../internal/styles/base");
const _componentstyles = require("../internal/styles/component-styles");
const _cssreset = require("../internal/styles/css-reset");
const _colors = require("../internal/styles/colors");
const _erroroverlay = require("../internal/components/errors/error-overlay/error-overlay");
const _devtoolsindicator = require("../internal/components/errors/dev-tools-indicator/dev-tools-indicator");
const _useerrorhook = require("../internal/container/runtime-error/use-error-hook");
function ReactDevOverlay(param) {
    let { state, globalError, children } = param;
    const [isErrorOverlayOpen, setIsErrorOverlayOpen] = (0, _react.useState)(false);
    const { readyErrors } = (0, _useerrorhook.useErrorHook)({
        errors: state.errors,
        isAppDir: true
    });
    const devOverlay = /*#__PURE__*/ (0, _jsxruntime.jsxs)(_shadowportal.ShadowPortal, {
        children: [
            /*#__PURE__*/ (0, _jsxruntime.jsx)(_cssreset.CssReset, {}),
            /*#__PURE__*/ (0, _jsxruntime.jsx)(_base.Base, {}),
            /*#__PURE__*/ (0, _jsxruntime.jsx)(_colors.Colors, {}),
            /*#__PURE__*/ (0, _jsxruntime.jsx)(_componentstyles.ComponentStyles, {}),
            /*#__PURE__*/ (0, _jsxruntime.jsx)(_devtoolsindicator.DevToolsIndicator, {
                state: state,
                readyErrorsLength: readyErrors.length,
                setIsErrorOverlayOpen: setIsErrorOverlayOpen
            }),
            /*#__PURE__*/ (0, _jsxruntime.jsx)(_erroroverlay.ErrorOverlay, {
                state: state,
                readyErrors: readyErrors,
                isErrorOverlayOpen: isErrorOverlayOpen,
                setIsErrorOverlayOpen: setIsErrorOverlayOpen
            })
        ]
    });
    return /*#__PURE__*/ (0, _jsxruntime.jsx)(_errorboundary.DevOverlayErrorBoundary, {
        devOverlay: devOverlay,
        globalError: globalError,
        onError: setIsErrorOverlayOpen,
        children: children
    });
}

if ((typeof exports.default === 'function' || (typeof exports.default === 'object' && exports.default !== null)) && typeof exports.default.__esModule === 'undefined') {
  Object.defineProperty(exports.default, '__esModule', { value: true });
  Object.assign(exports.default, exports);
  module.exports = exports.default;
}
