// next@15.2.0-canary.27/dist/client/components/react-dev-overlay/_experimental/pages/react-dev-overlay.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
Object.defineProperty(exports, "default", {
    enumerable: true,
    get: function() {
        return ReactDevOverlay;
    }
});
const _interop_require_wildcard = require("@swc/helpers/_/_interop_require_wildcard");
const _jsxruntime = require("react/jsx-runtime");
const _react = /*#__PURE__*/ _interop_require_wildcard._(require("react"));
const _shadowportal = require("../internal/components/shadow-portal");
const _base = require("../internal/styles/base");
const _componentstyles = require("../internal/styles/component-styles");
const _cssreset = require("../internal/styles/css-reset");
const _errorboundary = require("./error-boundary");
const _hooks = require("../../pages/hooks");
const _colors = require("../internal/styles/colors");
const _erroroverlay = require("../internal/components/errors/error-overlay/error-overlay");
const _devtoolsindicator = require("../internal/components/errors/dev-tools-indicator/dev-tools-indicator");
const _useerrorhook = require("../internal/container/runtime-error/use-error-hook");
function ReactDevOverlay(param) {
    let { children } = param;
    const { isMounted, state, onComponentError, hasRuntimeErrors, hasBuildError } = (0, _hooks.usePagesReactDevOverlay)();
    const { readyErrors } = (0, _useerrorhook.useErrorHook)({
        errors: state.errors,
        isAppDir: false
    });
    const [isErrorOverlayOpen, setIsErrorOverlayOpen] = (0, _react.useState)(true);
    return /*#__PURE__*/ (0, _jsxruntime.jsxs)(_jsxruntime.Fragment, {
        children: [
            /*#__PURE__*/ (0, _jsxruntime.jsx)(_errorboundary.DevOverlayErrorBoundary, {
                isMounted: isMounted,
                onError: onComponentError,
                children: children != null ? children : null
            }),
            isMounted && /*#__PURE__*/ (0, _jsxruntime.jsxs)(_shadowportal.ShadowPortal, {
                children: [
                    /*#__PURE__*/ (0, _jsxruntime.jsx)(_cssreset.CssReset, {}),
                    /*#__PURE__*/ (0, _jsxruntime.jsx)(_base.Base, {}),
                    /*#__PURE__*/ (0, _jsxruntime.jsx)(_colors.Colors, {}),
                    /*#__PURE__*/ (0, _jsxruntime.jsx)(_componentstyles.ComponentStyles, {}),
                    /*#__PURE__*/ (0, _jsxruntime.jsx)(_devtoolsindicator.DevToolsIndicator, {
                        state: state,
                        readyErrorsLength: readyErrors.length,
                        setIsErrorOverlayOpen: setIsErrorOverlayOpen
                    }),
                    (hasRuntimeErrors || hasBuildError) && /*#__PURE__*/ (0, _jsxruntime.jsx)(_erroroverlay.ErrorOverlay, {
                        state: state,
                        readyErrors: readyErrors,
                        isErrorOverlayOpen: isErrorOverlayOpen,
                        setIsErrorOverlayOpen: setIsErrorOverlayOpen
                    })
                ]
            })
        ]
    });
}

if ((typeof exports.default === 'function' || (typeof exports.default === 'object' && exports.default !== null)) && typeof exports.default.__esModule === 'undefined') {
  Object.defineProperty(exports.default, '__esModule', { value: true });
  Object.assign(exports.default, exports);
  module.exports = exports.default;
}
