// next@15.2.0-canary.27/dist/client/components/react-dev-overlay/_experimental/internal/components/errors/error-overlay/error-overlay.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
Object.defineProperty(exports, "ErrorOverlay", {
    enumerable: true,
    get: function() {
        return ErrorOverlay;
    }
});
const _jsxruntime = require("react/jsx-runtime");
const _builderror = require("../../../container/build-error");
const _errors = require("../../../container/errors");
const _rootlayoutmissingtagserror = require("../../../container/root-layout-missing-tags-error");
function ErrorOverlay(param) {
    let { state, readyErrors, isErrorOverlayOpen, setIsErrorOverlayOpen } = param;
    var _state_rootLayoutMissingTags;
    const isTurbopack = !!process.env.TURBOPACK;
    if (!!((_state_rootLayoutMissingTags = state.rootLayoutMissingTags) == null ? void 0 : _state_rootLayoutMissingTags.length)) {
        return /*#__PURE__*/ (0, _jsxruntime.jsx)(_rootlayoutmissingtagserror.RootLayoutMissingTagsError, {
            isTurbopack: isTurbopack,
            missingTags: state.rootLayoutMissingTags,
            versionInfo: state.versionInfo
        });
    }
    if (state.buildError !== null) {
        return /*#__PURE__*/ (0, _jsxruntime.jsx)(_builderror.BuildError, {
            isTurbopack: isTurbopack,
            message: state.buildError,
            versionInfo: state.versionInfo
        });
    }
    // No Runtime Errors.
    if (!state.errors.length) {
        return null;
    }
    if (!isErrorOverlayOpen) {
        return null;
    }
    return /*#__PURE__*/ (0, _jsxruntime.jsx)(_errors.Errors, {
        debugInfo: state.debugInfo,
        hasStaticIndicator: state.staticIndicator,
        isTurbopack: isTurbopack,
        errors: state.errors,
        readyErrors: readyErrors,
        versionInfo: state.versionInfo,
        onClose: ()=>{
            setIsErrorOverlayOpen(false);
        }
    });
}

if ((typeof exports.default === 'function' || (typeof exports.default === 'object' && exports.default !== null)) && typeof exports.default.__esModule === 'undefined') {
  Object.defineProperty(exports.default, '__esModule', { value: true });
  Object.assign(exports.default, exports);
  module.exports = exports.default;
}
