// next@15.2.0-canary.27/dist/client/components/react-dev-overlay/_experimental/internal/container/runtime-error/index.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
0 && (module.exports = {
    RuntimeError: null,
    styles: null
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    RuntimeError: function() {
        return RuntimeError;
    },
    styles: function() {
        return styles;
    }
});
const _tagged_template_literal_loose = require("@swc/helpers/_/_tagged_template_literal_loose");
const _jsxruntime = require("react/jsx-runtime");
const _react = require("react");
const _codeframe = require("../../components/code-frame/code-frame");
const _callstack = require("../../components/errors/call-stack/call-stack");
const _nooptemplate = require("../../helpers/noop-template");
const _componentstackpseudohtml = require("./component-stack-pseudo-html");
function _templateObject() {
    const data = _tagged_template_literal_loose._([
        "\n  ",
        "\n"
    ]);
    _templateObject = function() {
        return data;
    };
    return data;
}
function RuntimeError(param) {
    let { error } = param;
    const { firstFrame } = (0, _react.useMemo)(()=>{
        const firstFirstPartyFrameIndex = error.frames.findIndex((entry)=>!entry.ignored && Boolean(entry.originalCodeFrame) && Boolean(entry.originalStackFrame));
        var _error_frames_firstFirstPartyFrameIndex;
        return {
            firstFrame: (_error_frames_firstFirstPartyFrameIndex = error.frames[firstFirstPartyFrameIndex]) != null ? _error_frames_firstFirstPartyFrameIndex : null
        };
    }, [
        error.frames
    ]);
    return /*#__PURE__*/ (0, _jsxruntime.jsxs)(_jsxruntime.Fragment, {
        children: [
            firstFrame && /*#__PURE__*/ (0, _jsxruntime.jsx)(_codeframe.CodeFrame, {
                stackFrame: firstFrame.originalStackFrame,
                codeFrame: firstFrame.originalCodeFrame
            }),
            error.frames.length > 0 && /*#__PURE__*/ (0, _jsxruntime.jsx)(_callstack.CallStack, {
                frames: error.frames
            })
        ]
    });
}
const styles = (0, _nooptemplate.noop)(_templateObject(), _componentstackpseudohtml.PSEUDO_HTML_DIFF_STYLES);

if ((typeof exports.default === 'function' || (typeof exports.default === 'object' && exports.default !== null)) && typeof exports.default.__esModule === 'undefined') {
  Object.defineProperty(exports.default, '__esModule', { value: true });
  Object.assign(exports.default, exports);
  module.exports = exports.default;
}
